Sources/Exporters/OpenTelemetryProtocol/trace/OtlpHTttpTraceExporter.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import OpenTelemetrySdk
+import Foundation
+
+public func defaultOltpHttpTracesEndpoint() -> URL {
+    URL(string: "http://localhost:4318/v1/traces")!
+}
+
+public class OtlpHttpTraceExporter: SpanExporter {
+    let endpoint: URL
+    private let httpClient: HTTPClient
+    var pendingSpans: [SpanData] = []
+     
+    public init(endpoint: URL = defaultOltpHttpTracesEndpoint()) {
+        self.endpoint = endpoint
+        self.httpClient = HTTPClient()
+    }
+    
+    public func export(spans: [SpanData]) -> SpanExporterResultCode {
+        pendingSpans.append(contentsOf: spans)
+        return self.flush()
+    }
+
+    public func flush() -> SpanExporterResultCode {
+        var exporterResult: SpanExporterResultCode = .failure
+        let spans = pendingSpans
+        pendingSpans = []
+        let body = Opentelemetry_Proto_Collector_Trace_V1_ExportTraceServiceRequest.with {
+            $0.resourceSpans = SpanAdapter.toProtoResourceSpans(spanDataList: spans)
+        }
+        
+        var request = URLRequest(url: endpoint)
+        request.httpMethod = "POST"
+        request.httpBody = try? body.serializedData()
+        request.setValue("application/x-protobuf", forHTTPHeaderField: "Content-Type")
+                
+        httpClient.send(request: request) { result in
```

Comment:
Which thread does flush get called on? Given the amount of time it could take for a send to occur, are we at risk at blocking other work on same thread?

Reply:
@Sherlouk Thanks for the comments.  I am relatively new to Swift (I do have some background in Observability) so I really don't know.  Any suggestions you have on how to do this would be welcome.

Reply:
The flush method is explicitly called in the spanProcessor when called using forceFlush, and will execute on the calling thread (as documentation for Span Processor specifies), it is also called when explicitly calling shutdown().

Reply:
In this scenario, Nacho, flush is called on export within these new HTTP exporters so it will be the default.

I'm just not sure from memory what the threading is like within the module, and whether exports are already pushed onto a different thread or not. I'm conscious that URLSession is prone to blocking threads is all.

Reply:
True, the code is not very clear,  I missed it. Using flush in the name by default looks like synchronization after sending, we should not block by default.